text/0000-optional-jquery.md

```diff
@@ -0,0 +1,280 @@
+migration the addon should be placed in the default blueprint (until an eventual more aggressive deprecation of 
+jQuery). Only in the case the app is actively opting out of jQuery integration the addon is not needed.
+
+The addon itself has to make sure the Ember CLI version in use is at least the one that introduced the above mentioned
+logic, to prevent importing jQuery twice.
+
+### Assertions for jQuery based APIs
+
+Apart from testing (see below), Ember features some APIs that directly expose jQuery, which naturally cannot continue 
+to work without it. For these APIs some assertions have to be added when running in "no jQuery" mode (and not in 
+production), that provide some useful error messages for the developer:
+
+* `Ember.$()`
+  should throw an assertion stating that jQuery is not available.
+* `this.$()` in components
+  should throw an assertion stating that jQuery is not available and that `this.element` and native DOM APIs should be
+  used instead.
+
+### Introducing `ember-jquery-legacy` and deprecating `jQuery.Event` usage
+
+Event handler methods in components will usually receive an instance of [`jquery.Event`](http://api.jquery.com/category/events/event-object/) as an argument, which is very
+similar to native event objects, but not exactly the same. To name a few differences, not all properties of the native 
+event are mapped to the jQuery event, on the other hand a jquery event has a `originalEvent` property referencing the
+native event. 
+
+The updated event dispatcher in Ember 3.0 is capable of working without jQuery (similar to what 
+`ember-native-dom-event-dispatcher` provided for Ember 2.x). When jQuery is not available, it will naturally not be 
+able to pass a `jquery.Event` instance but a native event instead. This creates some ambiguity for addons, as they 
+cannot know in advance how the consuming app is built (with or without jQuery).
+
+For code that does not rely on any `jQuery.Event` specific API, there is no need to change anything as it will continue
+to work with native DOM events. 
+
+But there are cases where jQuery specific properties have to be used (when jQuery events are passed). This is especially
+true for the `originalEvent` property, for example to access `TouchEvent` properties that are not exposed on the 
+`jQuery.Event` instance itself. So there has to be a way to make the code work with either jQuery events or native 
+events being passed to the event handler (especially important for addons). Moreover this should be done in a way that 
+uses native DOM APIs only, to support the migration away from jQuery coupled code.
+
+To solve this issue another addon `ember-jquery-legacy` will be introduced, which for now will only expose a single
+`normalizeEvent` function. This function will accept a native event as well as a jQuery event (possibly distinguishing
+between those two modes at build time, based on the jQuery integration flag), but will always return a native event 
+only. 
```

Comment:
Will it be necessary for ember-jquery-legacy to be a dependency or can it be just a devdependency?

Reply:
Will have to be a dependency (for an addon), as `normalizeEvent` needs to be imported in actual code, see the little code example! Only needed though if you currently require any jQuery specifics like `originalEvent`.